src/lib/draw/route/RouteMode.svelte

```diff
@@ -61,11 +68,58 @@
       changeMode
     );
   });
+
+  async function fetchWithProgress(url, setProgress: (number) => void) {
+    const response = await fetch(url);
+    const reader = response.body.getReader();
+
+    const contentLength = parseInt(response.headers.get("Content-Length"));
+
+    let receivedLength = 0;
+    let chunks = [];
+    while (true) {
+      const { done, value } = await reader.read();
+      if (done) {
+        break;
+      }
+
+      chunks.push(value);
+      receivedLength += value.length;
+
+      const percent = (100.0 * receivedLength) / contentLength;
+      setProgress(percent);
+    }
+
+    let allChunks = new Uint8Array(receivedLength);
+    let position = 0;
+    for (let chunk of chunks) {
+      allChunks.set(chunk, position);
+      position += chunk.length;
+    }
+
+    return allChunks;
+  }
+
+  function routeToolInitialised() {
+    progress = 100;
+    routeToolReady = true;
+  }
 </script>
 
-{#if !routeTool}
-  <!-- TODO the text should be fixed, and the progress bar float -->
-  <div bind:this={progress}>Route tool loading...</div>
+{#if !routeToolReady && !failedToLoadRouteTool && !downloadComplete}
+  <label for="route-loading">Route tool loading</label>
```

Comment:
Elsewhere we've been doing `<label>Route tool loading <progress value={progress} />`, the second form mentioned by https://developer.mozilla.org/en-US/docs/Web/HTML/Element/label, to avoid making up an ID. Not too opinionated about it here